### ISSUE TITLE
Consider adding pgvector extension

### ISSUE THREAD

**Issue by ccjernigan** [fly-apps/postgres-flex#203]
[pgvector](https://github.com/pgvector/pgvector) enables vector similarity search which is useful for AI applications.  Support exists in Azure, AWS, and Supabase for this extension, indicating it is growing in popularity.

I had originally hoped to use Upstash Redis on Fly.io for a similarity search, but Upstash Redis does not support this functionality yet.  (Redis has the RediSearch extension for vector similarity search).

Pgvector support provides another way to achieve this.

**Comment by smorimoto** [comment]
That's not too hard to add on the end-user side. Shall I post an example `Dockerfile`? 🙂

**Comment by smorimoto** [comment]
I agree. That's why I think it makes the most sense to create a custom Dockerfile on the end-user side, build and deploy it as pg machine.

**Comment by tyrauber** [comment]
Hi @smorimoto, I'd love to see an example Dockerfile!  It would be super helpful to know how to customize the pg instance without having to fork the repo.

**Comment by lilouartz** [comment]
Has anyone actually managed to get it to work?
